### DISCUSSION TITLE
Custom TFM compat with net8.0 TFM

### DISCUSSION THREAD

**Discussion by Rosentti** [dotnet/msbuild#9756] [Q&A] [answered]
I define a custom TFM like so:
```xml
<!-- This is responsible for allowing us to define our custom native1.0 TFM -->

<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
    <PropertyGroup Condition="'$(TargetFramework)' == 'native1.0'">
        <TargetFrameworkIdentifier>Native</TargetFrameworkIdentifier>
        <TargetFrameworkVersion>v1.0</TargetFrameworkVersion>
        <TargetFrameworkProfile>Profile1</TargetFrameworkProfile>
    </PropertyGroup>
</Project>
```

And the import in my .proj (stripped for convenience)
```xml
<Import Project="Sdk.props" Sdk="Microsoft.Build.NoTargets" Version="3.7.0" />

<!-- The hook import -->
<Import Project="CustomTargetFrameworkParsingTarget.props"/>

<PropertyGroup>
        <TargetFramework>native1.0</TargetFramework>
</PropertyGroup>

<Import Project="Sdk.targets" Sdk="Microsoft.Build.NoTargets" Version="3.7.0" />
```

This works great, but I can't depend on the outputs of the .proj through a .csproj as it complains about incompatibilities:
`Project TestNative is not compatible with net8.0 (.NETCoreApp,Version=v8.0). Project TestNative supports: native10 (native,Version=v1.0)`

How could I make my native1.0 TFM compatible with net8.0?
I could also just use net8.0, but I'd like this to not be dependent on a .NET TFM.

**Comment by KalleOlaviNiemitalo** [comment]
I'm not sure about custom TFM, but other TFM conflicts can be forced with metadata like this:

```XML
<ItemGroup>
  <ProjectReference Include="../Tool/Tool.csproj">
    <SkipGetTargetFrameworkProperties>true</SkipGetTargetFrameworkProperties>
    <SetTargetFramework>TargetFramework=net462</SetTargetFramework>
  </ProjectReference>
</ItemGroup>
```

Perhaps it'd work with `<SetTargetFramework>TargetFramework=native1.0</SetTargetFramework>`, then.
According to <https://github.com/dotnet/sdk/issues/2280#issuecomment-392890515>, `SkipGetTargetFrameworkProperties` isn't even required.

NuGet might still have trouble with this, though.  <https://github.com/NuGet/Home/issues/5154>

**Comment by Rosentti** [comment]
Thanks. That got me in the right direction.
This works but is pretty suboptimal:
- Ideally I'd like to set any ReferenceOutputAssembly to false from TestProject.proj (eventually a NuGet package), so the projectreference would be cleaner. I only export "content" from TestProject.proj. 
- SetTargetFramework seems to be necessary unfortunately
```xml
<ProjectReference ReferenceOutputAssembly="False" SetTargetFramework="TargetFramework=native1.0" Include="..\TestProject\TestProject.proj" />
```

**Reply by Rosentti** [reply]
Making TestProject a `.nativeproj` made it possible to get rid of SetTargetFramework, but ReferenceOutputAssembly is still required.

**Reply by Rosentti** [reply]
I'd also be fine with the TestProject just having no targetframework (since there's only Content), but MSBuild really doesn't seem to like that.

**Comment by Rosentti** [comment, marked as answer]
Solved with 
```xml
<TargetFramework>native</TargetFramework>
``` 
and  making TestProject a .nativeproj.
`ReferenceOutputAssembly="False"` seems to always be necessary.
